### PR TITLE
Fix incorrect deletion of revNAT entries

### LINKED ISSUE #34411: Service ID conflict causes existing service IP to be disconnected after upgrade
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

equal or higher than v1.16.0 and lower than v1.17.0

### What happened?

Cilium 1.11 restores services from `cilium_lb4_backends_v2` map. cilium 1.12 and later versions restore services from `cilium_lb4_services_v2` map. a service fails to be restored due to service ID has been registered after cilium-agent upgrade 1.12, one of the conflicting services will be deleted because the k8s service no longer exists, which will cause the existing service IP to be unreachable, such as k8s service ClusterIP. 

Through `cilium monitor` capture packet , it was found that when the lxc NIC processed the `to-container` traffic, it did not convert the service backend IP into the service frontend IP.

### How can we reproduce the issue?

1.Use bpftool to insert a conflicting service ID entry in `cilium_lb4_services_v2` map
2.Restart cilium-agent, and ping service IP from the container

### Cilium Version

1.12.19

### Kernel Version

5.15.80

### Kubernetes Version

1.28.3

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by haozhangami**
If there is a service ID conflict in the `cilium_lb4_services_v2` map
and one of the services no longer exists, the no longer present service
will be deleted after the k8s services are synchronized.The revNAT entry
of the corresponding service is accidentally deleted, causing the other
existing service IP to be disconnected.
    
This patch checks the consistency of the service IP:Port in service and
revNat maps. If they are inconsistent, the inconsistent services will be
deleted.

Fixes: #34411

```release-note
Fix incorrect deletion of revNAT entries due to service ID conflict
```

### CODE REVIEW

**Comment by christarazi** [comment]
I don't think I'm familiar enough with this code to properly review this PR. Could someone from @cilium/sig-lb take a look?

**Comment by haozhangami** [comment]
@cilium/sig-lb LB experts, Please take a look.

**Comment by haozhangami** [comment]
Hi @christarazi @joestringer, Can you assign a LB expert to take a look?

**Comment by joestringer** [comment]
Cilium is an open source community of volunteers who are interested in making the shared implementation better. I can CC @cilium/sig-lb but I have no control to direct or assign people to accelerate review. I agree with you though that taking three weeks to provide feedback on the PR feels too long; this PR seems like it will help to resolve some bugs for users. On the other hand, I know that NAT is a complex topic and it's very easy to break something while fixing something else, so I would like to see the review to pay close attention to what's changing to ensure that we think this is the right fix.

**Comment by joestringer** [comment]
Maybe something that could help the process along would be if there was an automated test case that demonstrates the change in behavior and protects against regressions? At the moment it looks like this change consists of the fix only with no tests that demonstrate the original bug or how this fixes it.

**Comment by joamaki** [comment]
Hey, took a look and in its current form I would not be happy with merging this.

Are there other reasons for this fix besides the v1.11 map migration issue? I don't think that alone warrants this.
Can you for example come up with a scenario where the latest version of Cilium could end up with these maps being out of sync?

I would be more open to a "consistency check" during restoration to resolve invalid map state. E.g. during map dumping validate that the services and revnat maps are consistent to each other and remove both the service and revnat if they are not. This way the normal path would not need to deal with this situation.

**Comment by haozhangami** [comment]
@joestringer @joamaki Thank you for your reply. 

I reproduce this problem on cilium 1.16.1. I record the reproduction steps in the issue. The cost of comparing all service IPs in the two maps may be higher than the current way. I don’t have a good idea at the moment, I will try to check the consistency during the restoration.

If you have time, please take a look. Looking forward to your suggestions.

**Comment by haozhangami** [comment]
Hi @joamaki,  I update the code to check the consistency of the IP:Port in service and revnat maps. Please take a look, thanks!

**Comment by haozhangami** [comment]
Conformance Runtime failure is due to a known flake #34731
https://github.com/cilium/cilium/actions/runs/11016800229/job/30593327582
Conformance IPsec E2E failure is due to a known flake #33084
https://github.com/cilium/cilium/actions/runs/11016799131/job/30593205064
Integration Test (ubuntu-22.04-arm64) failure seems to be unable to connect to etcd
https://github.com/cilium/cilium/actions/runs/11016800210/job/30593190923
Conformance Ginkgo (ci-ginkgo) failure is due to endpoint regeneration failed
https://github.com/cilium/cilium/actions/runs/11016799694/job/30593334921

Hi @pchaigno, Could you please rerun the failing tests?

File: pkg/maps/lbmap/lbmap.go [hunk 1]
```diff
@@ -505,12 +531,25 @@ func (*LBBPFMap) DumpServiceMaps() ([]*loadbalancer.SVC, []error) {
...
+		if err != nil {
+			errors = append(errors, err)
+		}
 		err = Service6MapV2.DumpWithCallback(parseSVCEntries)
 		if err != nil {
 			errors = append(errors, err)
 		}
 	}
 
+	for _, svcKey := range inconsistentServiceKeys {
```
Comment (joamaki) [inline pkg/maps/lbmap/lbmap.go:544]: `DumpServiceMaps` as the name implies should be just dumping them and not deleting entries. Please move this into a separate method. Other than that this seems like a reasonable approach to me!

**Review by joamaki — APPROVED** [review]
Seems like an ok approach to me. It would be good to add tests for this though, but I'm ok with merging it in this state.

**Pull request merged**